src/smartcontracts/interface.ts

```diff
@@ -16,7 +15,7 @@ export interface ISmartContract {
     /**
      * Gets the address of the Smart Contract.
      */
-    getAddress(): Address;
+    getAddress(): IBech32Address;
```

Comment:
Q - why do we have IBech32 as opposed to an interface for the address?

Reply:
It was more like a choice to avoid ambiguity. That is, `IBech32Address` only has the `bech32()` method, without `pubkey()` and `hex()`, being thus being a bit more "portable" between different domains (e.g. `walletcore`, `network-providers` and so on).

We can change this, though. `IAddress`?

Reply:
Renamed to `IAddress`.